docs/tutorials/ruby-driver-crud-operations.txt

```diff
@@ -458,20 +458,67 @@ the modification occurs.
   doc = artists.find_one_and_update( { :name => 'José James' }, { '$set' => { :name => 'José' } } )
   doc # Return the document before the update.
 
-``find_one_and_replace``
+Update Options
+~~~~~~~~~~~~~~
+
+To add options to an update command, specify them as key value pairs in the options
+Hash argument.
 
 .. code-block:: ruby
 
-  doc = artists.find(:name => 'José James').
-    find_one_and_replace( { '$set' => { :name => 'José' } }, :return_document => :after )
-  doc # Return the document after the update.
+  client = Mongo::Client.new([ '127.0.0.1:27017' ], :database => 'music')
+  artists = client[:artists]
 
-  doc = artists.find_one_and_replace(
-    { :name => 'José James' },
-    { '$set' => { :name => 'José' } },
-    :return_document => :after
+  # Use the _id index to perform this operation
+  result = artists.update_one(
+    { :name => 'Goldie' },
+    { "$inc" => { :plays =>  1 } },
+    { hint: { _id: 1 } }
```

Comment:
Is _id index used for the update rather than the find, i.e. this call searches for the document twice?

Reply:
This call doesn't search for documents twice! My understanding is that `find` just creates a `View` on the collection, and then the update command performs the search (with the specified index) and the update.

Reply:
In this case I am confused: how would the _id index be used with a query on name?

Reply:
I'll think of a clearer example.